### PR TITLE
Added subrepo name rule to interpreter

### PR DESCRIPTION
**Pull request by Tatskaari**
I wonder if cononicalise could be made to handle sub-repos as well. 

fixes: #1179

### CODE REVIEW

File: src/parse/asp/builtins.go [hunk 1]
```diff
@@ -578,6 +579,10 @@ func packageName(s *scope, args []pyObject) pyObject {
 	return pyString(s.pkg.Name)
 }
 
+func subrepoName(s *scope, args []pyObject) pyObject {
+	return pyString("///" + s.pkg.SubrepoName)
```
Comment (sagikazarmark) [inline src/parse/asp/builtins.go:583]: Note that `package_name` does not. Not sure if that's an inconsistency you want to care about, but it's worth noting.

Also, what if there is no subrepo name?

Ideally, I would use it like this: `subrepo_name() + package_name()` or `subrepo_name() + "//tools/k8s:helm"`.

If I want it to work in both subrepos and non-subrepos, I have to check if `subrepo_name() == "///"`.

**Review by sagikazarmark — COMMENTED** [review]
> I wonder if cononicalise could be made to handle sub-repos as well.

I think that would be great. If `canonicalise("//tools/k8s:helm")` would return `//tools/k8s:helm` in the same repo and `///subrepo//tools/k8s:helm` when used as a subrepo, that would be the best.

File: src/parse/asp/builtins.go (hunk 1 above)
Comment (Tatskaari) [reply src/parse/asp/builtins.go:583]: Haha my excuse is that this was a last minute PR at 7pm on a Monday.

Comment (sagikazarmark) [reply src/parse/asp/builtins.go]: No worries! Thanks a lot!

**Review by sagikazarmark — APPROVED** [review]

**Pull request merged**